src/Microsoft.Identity.Client/MsalServiceException.cs

```diff
@@ -221,6 +221,16 @@ public MsalServiceException(string errorCode, string errorMessage, int statusCod
         /// </remarks>
         public HttpResponseHeaders Headers { get; internal set; }
 
+        /// <summary>
+        /// A string that provides more details about the error. Sent by the server.
+        /// </summary>
+        public string SubError { get; internal set; }
+
+        /// <summary>
+        /// An ID that can used to piece up a single authentication flow.
+        /// </summary>
+        public string CorrelationId { get; internal set; }
```

Comment:
Are we sure we should expose this here?

Reply:
Bogdan and I were wondering about that. It was exposed in ADAL IFAIK.

Reply:
if that's the case I'm ok. Not sure it's required though should be in the logs, not sure what people will get out of us exposing it

---
In reply to: [251521718](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/841#discussion_r251521718) [](ancestors = 251521718)

Reply:
I would suggest to not expose it. It's an internal concept right?

---
In reply to: [251524638](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/841#discussion_r251524638) [](ancestors = 251524638,251521718)